### PR TITLE
search_gui_close_search(): avoid crash when emptying search list twice.

### PR DESCRIPTION
**Pull request by marcodiegomesquita**
It looks like notebook_search_results should never be emptyed. We can
avoid it being emtpy handlig this special case in search_gui_close_search
by insensityzing current search tree and calling
search_gui_switch_search(NULL).

### CODE REVIEW

**Comment by rmanfredi** [comment]
@marcodiegomesquita thank you so much for the fix in callback.c.  I've been trying to locate the source of this bug for a long time and I am glad you found it: it was a major holding for the new 1.1.16 release!

File: src/ui/gtk/search_common.c [hunk 1]
```diff
@@ -716,6 +716,10 @@ search_gui_clear_results(void)
 	search_gui_update_status(search);
 }
 
+
+static void
```
Comment (rmanfredi) [inline src/ui/gtk/search_common.c:720]: Minor style issue: better spell-out forward declarations on a line by itself, like:

`static void search_gui_switch_search(struct search *search);`

and reserve the formatting you used to the actual definition of the routine.

Comment (marcodiegomesquita) [reply src/ui/gtk/search_common.c:720]: Will you take care o this? Should I "commit --amend" then "push -f" it? Open a new pull request?

Comment (rmanfredi) [reply src/ui/gtk/search_common.c:720]: If you are not expert with git, I can take care of it after merging your commits in gtk-gnutella.
Or you can cleanup on your local branch by adding an extra commit.

If you are eager to play, you can rewrite yiour history with "git rebase -i HEAD^^^" and then move your last commit up and request a "fixup" merging with the original commit.  I have never done that on pull requests, so I have no idea what is going to happen with GitHub here if you fix your local history and force-push the changes to the pull request.

Let me know how you wish to proceed :-)

Comment (marcodiegomesquita) [reply src/ui/gtk/search_common.c]: I added one more commit fixing it.

**Comment by rmanfredi** [comment]
Excellent, thank you!

**Pull request merged**